Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/squash.py

```diff
@@ -0,0 +1,166 @@
+        if result.returncode:
+            sys.stderr.write(result.stderr)
+            sys.stderr.write('Failed to get all commits from HEAD to {}'.format(base_commit.hash))
+            return None
+        return result.stdout.decode('utf-8').strip().splitlines()
+
+    @classmethod
+    def undo_reset(cls, repository):
+        return run([repository.executable(), 'reset', "'HEAD@{1}'"], cwd=repository.root_path)
+
+    @classmethod
+    def squash_commit(cls, args, repository, branch_point, **kwargs):
+        # Make sure we have the commit that user want to revert
+        try:
+            if args.base_commit:
+                base_commit = repository.find(args.base_commit, include_log=True)
+                if hasattr(base_commit, 'identifier') and base_commit <= branch_point:
+                    sys.stderr.write('It seems you are trying to sqaush beyond branch point.')
+                    return 1
+            else:
+                base_commit = branch_point
+            # Check if there are any outstanding changes:
+            if repository.modified():
+                sys.stderr.write('Please commit your changes or stash them before you squash to base commit: {}'.format(base_commit))
+                return 1
+        except (local.Scm.Exception, ValueError) as exception:
+            # ValueErrors and Scm exceptions usually contain enough information to be displayed
+            # to the user as an error
+            sys.stderr.write('Could not find "{}"'.format(args.base_commit) + '\n')
+            return 1
+
+        if args.interactive:
+            result = run([repository.executable(), 'rebase', '-i'] + [base_commit.hash], cwd=repository.root_path)
+        else:
+            previous_history = ''
+            commit_hash_list = cls.get_commits_hashes(repository, base_commit)
+            if not commit_hash_list:
+                return 1
+            if not args.no_sub_commit_message:
+                commits = map(lambda hash: repository.find(hash, include_log=True), commit_hash_list)
+                previous_history += '\n'.join(map(lambda commit: commit.message, commits))
+            result = run([repository.executable(), 'reset'] + [base_commit.hash], cwd=repository.root_path)
+            if result.returncode:
+                sys.stderr.write('Failed to merge the diff.')
+                cls.undo_reset()
+            modified_files = repository.modified()
+            if not modified_files:
+                sys.stderr.write('Failed to detect any diff to merge.')
+                return 1
+            for file in modified_files:
+                if run([repository.executable(), 'add', file], cwd=repository.root_path).returncode:
+                    sys.stderr.write("Failed to add '{}'\n".format(file))
+                    return 1
+            env = os.environ
+            env['COMMIT_MESSAGE_CONTENT'] = previous_history
```

Comment:
If I'm understanding this line correctly, we're populating the commit message by concatenating the existing commit messages, right?